src/integer_mod_q/mat_zq/arithmetic/mul_scalar.rs

```diff
@@ -0,0 +1,369 @@
+    ///
+    /// Parameters:
+    /// - `scalar`: specifies the scalar by which the matrix is multiplied
+    ///
+    /// Returns the product of `self` and `scalar` as a [`MatZq`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{MatZq, Zq};
+    /// use std::str::FromStr;
+    ///
+    /// let mat1 = MatZq::from_str("[[42, 17],[8, 6]] mod 61").unwrap();
+    /// let integer = Zq::try_from((2, 61)).unwrap();
+    ///
+    /// let mat2 = &mat1 * &integer;
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Panics if the moduli mismatch.
+    fn mul(self, scalar: &Zq) -> Self::Output {
+        self.mul_scalar_safe(scalar).unwrap()
+    }
+}
+
+arithmetic_trait_reverse!(Mul, mul, Zq, MatZq, MatZq);
+
+arithmetic_trait_borrowed_to_owned!(Mul, mul, MatZq, Zq, MatZq);
+arithmetic_trait_borrowed_to_owned!(Mul, mul, Zq, MatZq, MatZq);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, MatZq, Zq, MatZq);
+arithmetic_trait_mixed_borrowed_owned!(Mul, mul, Zq, MatZq, MatZq);
+
+impl MatZq {
+    /// Implements multiplication for a [`MatZq`] matrix with a [`Zq`].
+    ///
+    /// Parameters:
+    /// - `scalar`: specifies the scalar by which the matrix is multiplied
+    ///
+    /// Returns the product of `self` and `scalar` as a [`MatZq`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::{MatZq, Zq};
+    ///  use std::str::FromStr;
+    ///
+    /// let mat1 = MatZq::from_str("[[42, 17],[8, 6]] mod 61").unwrap();
+    /// let integer = Zq::try_from((2, 61)).unwrap();
+    ///
+    /// let mat2 = &mat1.mul_scalar_safe(&integer);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type
+    /// [`MathError::MismatchingModulus`] if the moduli mismatch.
+    pub fn mul_scalar_safe(&self, scalar: &Zq) -> Result<Self, MathError> {
+        if self.get_mod() != scalar.modulus {
+            return Err(MathError::MismatchingModulus(format!(
+                " Tried to add matrixes with moduli '{}' and '{}'.",
+                self.get_mod(),
+                scalar.modulus
+            )));
+        }
+
+        let mut out =
+            MatZq::new(self.get_num_rows(), self.get_num_columns(), self.get_mod()).unwrap();
+        unsafe {
+            fmpz_mod_mat_scalar_mul_fmpz(&mut out.matrix, &self.matrix, &scalar.value.value);
+        }
+        Ok(out)
+    }
+}
+
+#[cfg(test)]
+mod test_mul_z {
+
+    use crate::integer::Z;
+    use crate::integer_mod_q::MatZq;
+    use std::str::FromStr;
+
+    /// Checks if matrix multiplication works fine for both borrowed
+    #[test]
+    fn borrowed_correctness() {
+        let mat1 = MatZq::from_str("[[42, 17],[8, 6]] mod 61").unwrap();
+        let mat2 = mat1.clone();
+        let mat3 = MatZq::from_str("[[84, 34],[16, 12]] mod 61").unwrap();
+        let integer = Z::from(2);
+
+        let mat1 = &mat1 * &integer;
+        let mat2 = &integer * &mat2;
+
+        println!("{}", mat1);
```

Comment:
```suggestion
        let mat1 = &mat1 * &integer;
        let mat2 = &integer * &mat2;
```
And for the next time, probably do not reuse the same variable name, but choose another one for the results as these clones would not be required otherwise (in several test functions)